integration-test/src/main/java/com/sequenceiq/it/cloudbreak/BlueprintTests.java

```diff
@@ -3,39 +3,269 @@
+import org.slf4j.LoggerFactory;
 import org.springframework.util.StreamUtils;
 import org.testng.Assert;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeTest;
 import org.testng.annotations.Test;
 
+import javax.ws.rs.BadRequestException;
 import java.io.IOException;
 
 public class BlueprintTests extends CloudbreakTest {
 
-    public static final String UPLOADBP = "bluuvelvet";
+    public static final String VALID_BP_NAME = "valid-blueprint";
+
+    public static final String AGAIN_BP_NAME = "again-blueprint";
+
+    public static final String LONG_DC_BP_NAME = "long-description-blueprint";
+
+    public static final String DELETE_BP_NAME = "delete-blueprint";
+
+    public static final String SPECIAL_BP_NAME = "@#$%|:&*;";
+
+    public static final String INVALID_SHORT_BP_NAME = "";
+
+    public static final String INVALID_LONG_BP_NAME = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+
+    public static final String EMPTY_BP_NAME = "temp-empty-blueprint";
+
+    public static final String INVALIDURL_BP_NAME = "temp-url-blueprint";
+
+    public static final String BP_DESCRIPTION = "temporary blueprint for API E2E tests";
+
+    public static final String INVALID_LONG_DESCRIPTION = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudbreakTest.class);
+
+    private String errorMessage = "";
 
     @Test
-    public void createBlueprint22() throws Exception {
+    public void createValidBlueprint() throws Exception {
         given(CloudbreakClient.isCreated());
         given(Blueprint.request()
-                .withName(UPLOADBP)
-                .withDescription("alma")
+                .withName(VALID_BP_NAME)
+                .withDescription(BP_DESCRIPTION)
                 .withAmbariBlueprint(getBlueprintFile()));
         when(Blueprint.post());
         then(Blueprint.assertThis(
-                (blueprint, t)-> {
-                    Assert.assertNotNull(blueprint.getRequest());
+                (blueprint, t) -> {
+                    Assert.assertEquals(blueprint.getResponse().getName(), VALID_BP_NAME);
+                })
+        );
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
```

Comment:
Expected exception is not the bet way to handle errors in test. Would be better to use `org.junit.rules.ExpectedException`

Reply:
This is the [TestNG way](http://testng.org/doc/documentation-main.html#annotations) for expect and exception.

Reply:
Up to tou darling. Personally i fixed 50 tests which didn't test what it wanted because the exception occurred earlier than the test expected. On the other hand parsing text during a nightly test is not the best solution.

Reply:
I can only cook from what I got :smiley: